crates/sui-bcs-json/README.md

```diff
@@ -0,0 +1,71 @@
+# Description
+
+This library provides an interface for transforming BCS data into JSON and vice-versa. The aim is to provide a way to support BCS in an interoperable manner for languages that do not have native BCS support, but have C interoperability.
+
+# Design choices
+
+The library was designed to be as simple as possible to allow users to convert to JSON or to BCS by passing the `typename` (as BCS requires to know before hand the type of the encoded data). Specifically, the types that are supported to be converted as of now are the `TransactionKind` and its subtypes from the main `Sui Transaction` type [transaction.rs](https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/transaction.rs)
+
+## API
+The library offers two functions for serializing and deserializing data:
+* **sui_bcs_from_json**
+```c
+size_t sui_bcs_from_json(const char *type_name,
+                         const char *json_data,
+                         const uint8_t **result,
+                         size_t *length);
+```
+
+This function **sui_bcs_from_json** -- will transfrom a JSON string input into a BCS array (u8). The function takes as input a typename (`type_name`), the input json string (`json_data`), a pointer to a pointer where to store the results (`result`), and a pointer to store the length of the result. The function returns 0 for success, 1 or 2 for failure. 
+
+
+* **sui_bcs_to_json**
+
+```c
+size_t sui_bcs_to_json(const char *type_name,
+                       const uint8_t **bcs_buf_ptr,
+                       size_t len,
+                       const char **result,
+                       bool pretty);
+  
+```
+
+
+This function **sui_bcs_to_json** will transform a BCS array input into a JSON string. The function takes as input a typename (`type_name`), the input BCS array (`bcs_buf_ptr`) as a pointer, a pointer to the length of the result (`length`), and a boolean flag if the JSON output should be pretty printed or not (`pretty`).
+
+
+## Memory management
+There are two functions that should be used to free Rust allocated data (`Vec<u8>`` and `String`):
+
+```c
```

Comment:
We should consider merging this into one freeing function, because both types here are basically the same thing.  This would require us being able to always recover the length of the allocation from a pointer to it (similar to how `malloc` and `free` work).

Reply:
That makes sense, fixed!